spk/sonarr3/src/service-setup.sh

```diff
@@ -0,0 +1,90 @@
+PATH="${SYNOPKG_PKGDEST}/bin:${PATH}"
+MONO_PATH="/var/packages/mono/target/bin"
+MONO="${MONO_PATH}/mono"
+SONARR="${SYNOPKG_PKGDEST}/share/Sonarr/Sonarr.exe"
+
+# Sonarr uses custom Config and PID directories
+HOME_DIR="${SYNOPKG_PKGVAR}"
+CONFIG_DIR="${SYNOPKG_PKGVAR}/.config"
+SONARR_CONFIG_DIR="${CONFIG_DIR}/Sonarr"
+PID_FILE="${SONARR_CONFIG_DIR}/sonarr.pid"
+
+# Some have it stored in the root of package
+LEGACY_CONFIG_DIR="${SYNOPKG_PKGDEST}/.config"
+
+# workaround for mono bug with armv5 (https://github.com/mono/mono/issues/12537)
+if [ "$SYNOPKG_DSM_ARCH" == "88f6281" ] || [ "$SYNOPKG_DSM_ARCH" == "88f6282" ]; then
+    MONO="MONO_ENV_OPTIONS='-O=-aot,-float32' ${MONO}"
+fi
+
+# for DSM < 7 only:
+GROUP="sc-download"
+LEGACY_GROUP="sc-media"
+
+SERVICE_COMMAND="env PATH=${MONO_PATH}:${PATH} HOME=${HOME_DIR} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${MONO} ${SONARR}"
+SVC_BACKGROUND=y
+SVC_WAIT_TIMEOUT=90
+
+validate_preinst ()
+{
+    # check if the installed distribution is a legacy version
+    if [ -f "${SYNOPKG_PKGDEST}/share/NzbDrone/NzbDrone.exe" ]; then
+        # v2 installed
+        echo "Update from NzbDrone (Sonarr v2.x) is not supported."
+        exit 1
+    fi
+}
+
+service_postinst ()
+{
+    echo "Set update required"
+    # Make Sonarr do an update check on start to avoid possible Sonarr
+    # downgrade when synocommunity package is updated
```

Comment:
As we now save and restore the current installation, this comment is outdated. Propose to comment like this:
```suggestion
    # Make Sonarr do an update check on start to update to the latest version available
```